packages/utils/src/lib/report.ts

```diff
@@ -161,6 +167,103 @@ export function countCategoryAudits(
   }, 0);
 }
 
+export function getAuditsFromAllPlugins(
+  report: ScoredReport,
+): EnrichedAuditReport[] {
+  return report.plugins.reduce<EnrichedAuditReport[]>(
+    (acc, plugin) => [...acc, ...plugin.audits],
+    [],
+  );
+}
+
+export function getAuditByRef(
+  { slug, weight, plugin }: CategoryRef,
+  plugins: ScoredReport['plugins'],
+): WeighedAuditReport {
+  const auditPlugin = plugins.find(
+    ({ slug }) => slug === plugin,
+  ) as PluginReport;
+  const audit = auditPlugin?.audits.find(
+    ({ slug: auditSlug }) => auditSlug === slug,
+  );
+  if (!audit) {
+    throwIsNotPresentError(`Audit ${slug}`, auditPlugin?.slug);
+  }
+  return {
+    ...audit,
+    weight,
+    plugin,
+  };
+}
+
+export function getGroupWithAudits(
+  refSlug: string,
+  refPlugin: string,
+  plugins: ScoredReport['plugins'],
+): EnrichedScoredAuditGroupWithAudits {
+  const plugin = plugins.find(({ slug }) => slug === refPlugin) as PluginReport;
+  const groupWithAudits = plugin?.groups?.find(
+    ({ slug }) => slug === refSlug,
+  ) as EnrichedScoredAuditGroupWithAudits;
```

Comment:
Same here. Using `as T` is just an alternative syntax for `!` in these cases, BTW. I see it defeats our linter, but it's best avoided for the exact same reasons. (I'll look around for a lint rule that covers this :thinking:)

```suggestion
  const plugin = plugins.find(({ slug }) => slug === refPlugin);
  const groupWithAudits = plugin?.groups?.find(
    ({ slug }) => slug === refSlug,
  );
```

Reply:
You're correct, removed